Ender2-Pro

spuder opened the pull request:
Adds Ender2 pro config. Copied from ender3 v2

The [Ender 2 pro](https://www.creality3dofficial.com/products/creality-ender-2-3d-printer) is a new printer from creality that uses the Creality Silent Motherboard version 4.2.3 

The only real difference between an Ender 3 (using 4.2.2. board) and Ender2-Pro (4.2.3 board) is the following pins have been changed

| PIN | 4.2.2 | 4.2.3| 
| --- | --- | ---| 
| Heater Bed Pin | PA2 | PB10 | 
| Button Encoder 1 | PB10  |PA2|

# Resources
- https://github.com/MarlinFirmware/Marlin/pull/23307
- https://github.com/MarlinFirmware/Configurations/pull/633

Marking as 'Draft' as this hasn't yet been tested

KevinOConnor commented:
I'm not sure what the state of this PR is.  At a minimum commits would need to have a signed-off-by line containing your full real name ( https://www.klipper3d.org/CONTRIBUTING.html ).

-Kevin

spuder commented:
I've updated against master and added a signed-off-line commit. 
I still have not tested this since my printer is still running Marlin, and I most likely won't have time to convert it any time soon. 

We can either merge this as-is and fix any issues that pop up, 
Or wait until we have someone test this more thoroughly.

spuder commented:
I've tried testing, unfortunately I'm currently stuck and unable to flash the klipper firmware to the 4.2.3 board using [these instructions](https://www.klipper3d.org/Installation.html)

Leaving a note here in case anyone else is able to have better luck

```
# Get serial device
ls /dev/serial/by-id/*
/dev/serial/by-id/usb-1a86_USB_Serial-if00-port0

sudo service klipper stop

Version: v0.10.0-312-g81283423
  Linking out/klipper.elf
  Creating hex file out/klipper.elf.hex
  Flashing out/klipper.elf.hex to /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0 via avrdude
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 3 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 4 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 5 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 6 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 7 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 8 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 9 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 10 of 10: not in sync: resp=0x00

avrdude done.  Thank you.

make: *** [src/avr/Makefile:34: flash] Error 1
```


It seems others have had similar problems https://github.com/Klipper3d/klipper/issues/1632
I've also followed the support article here: https://github.com/Klipper3d/klipper/blob/master/docs/FAQ.md#the-make-flash-command-doesnt-work
